### DISCUSSION TITLE
Welcome to OnlyHumans-Proxy Discussions!

### DISCUSSION THREAD

**Discussion by pulkitsharma07** [pulkitsharma07/OnlyHumans-Proxy#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ctrain79** [comment]
This idea is wild---and so many implications.

Glaze and watermarking could be used. Steganography could be used. Hackers and Security Industry would have endless advances iterating on this for sure.

Perhaps piece-wise images, rather than larger chunks---say, like a sprite atlas for reusable parts to reduce the memory involved and allow for some reflow capability. Flow and scaling for the page from user configuration would be a nightmare, I think, but why not tackle that.

**Reply by pulkitsharma07** [reply]
Yeah, smaller chunks will definitely help. Different chunks can refresh independently at different refresh rates as and when needed.

I was thinking that users can use `iframes` to do "chunking" on their frontend, but I understand what you mean, if this proxy somehow  figures that out by default, that can improve the developer experience.

Just thinking out loud, instead of rendering the full page as an image, how about I extract particular `<div>`, `<p>` or `<h1>` tags and just render those as images, because it is the text content which we want to defend from bots. Rest of the page can remain unchanged, so interactivity via JS or HTML should continue to work.
